File: backend/edugate-core/src/main/java/de/svws_nrw/edugate/core/domain/Schule.java
package de.svws_nrw.edugate.core.domain;

import java.time.Instant;
import java.util.UUID;

/**
 * Eine Schule eines Schulträgers (Mandant: {@code schultraegerId}).
 *
 * @param quelle Herkunft gemäß ADR-020: {@code LANDESLISTE}, {@code MANUELL} oder {@code SONDERFALL}
 */
public record Schule(
    UUID id,
    UUID schultraegerId,
    String schulnummer,
    String name,
    boolean aktiv,
    UUID katalogId,
    String quelle,
    String sonderfallHinweis,
    Instant createdAt,
    Instant updatedAt
) {
}

File: frontend/src/types/schule.ts
/** Herkunft einer Schule (ADR-020). */
export type SchuleQuelle = 'LANDESLISTE' | 'MANUELL' | 'SONDERFALL'

export interface Schule {
  id: string
  schultraegerId: string
  schulnummer: string
  name: string
  aktiv: boolean
  katalogId: string | null
  quelle: SchuleQuelle
  sonderfallHinweis: string | null
  createdAt: string
  updatedAt: string
}

export interface SchuleFormData {
  schulnummer: string
  name: string
}

/**
 * Formulardaten beim Neuanlegen (ADR-020): katalogId/sonderfall/sonderfallHinweis gibt es nur
 * hier, nicht beim Bearbeiten - die Herkunft einer Schule wird beim Anlegen einmalig festgelegt
 * und bleibt danach unverändert (analog SchultraegerCreateFormData).
 */
export interface SchuleCreateFormData extends SchuleFormData {
  katalogId?: string
  sonderfall?: boolean
  sonderfallHinweis?: string
}
